Add option to hide/show main toolbar

xiota opened the pull request:
It's currently possible to hide the main toolbar using its context menu.  However, to re-show it requires using the context menu from the annotation toolbar.  This makes re-showing the main toolbar a multi-step process (show annotation bar, context menu, show main toolbar, re-hide annotation bar).  This pull request adds the option to hide/show the main toolbar directly.

tsujan reviewed (approved):
GTM. Thanks!

@stefonarch, would you please test it too? The behavior is like the annotation toolbar: showing/hiding of the main toolbar shouldn't be remembered but there's an option for it in Preferences dialog.

stefonarch commented:
> (show annotation bar, context menu, show main toolbar, re-hide annotation bar)

Maybe I'm missing something, but to me steps are : right click besides menu texts to show context menu > select show toolbar and/or annotation bar:

![schermata-09-02-07-37-01](https://user-images.githubusercontent.com/10681413/131787991-82ca8de2-7035-4643-9445-540ccba42a39.png)

So I cannot notice any change now...

xiota commented:
I'm using a global menubar, so I don't have that context menu on the menubar.  Also, some people might like the toolbar hidden by default on startup (option in preferences).

xiota commented:
I made some more changes to preferences.  Changed the "Thumbnails" tab to "Dock".  Added options to show Exif data on startup, change position of the Thumbnails dock, and change the size of thumbnails.   Should I add it to this PR, or open up a new one?  Here's a screenshot:

![screenshot-01-dock](https://user-images.githubusercontent.com/8353098/131827412-78996d19-72ac-406f-8549-ba87a0308837.png)

tsujan commented:
> Should I add it to this PR

Please don't do that! Each PR should be about a single change.

tsujan commented:
@stefonarch 
Yes, the toolbar can be shown/hidden by right clicking on the menu-bar too but I see this PR as a way of treating the main toolbar like the annotation toolbar and having an option for hiding it with startup.

xiota commented:
@tsujan So... I should submit a separate pull request for each option?

tsujan commented:
>  I should submit a separate pull request for each option?

Yes, please. Each PR should be about a single subject. The current PR is good as it is. It's about the main toolbar.

Please also note that the second tab of Preferences is only about thumbnails. I don't think we should add exif data to it but we could discuss it when you make your next PR.

tsujan commented:
I merge this because its code is clean and it treats both toolbars in the same way.
